generic_scripts/nlk.py

```diff
@@ -0,0 +1,290 @@
+import numpy as _np
+from numpy.polynomial.polynomial import polyfit as _polyfit
+import matplotlib.pyplot as _plt
+from mathphys.constants import vacuum_permeability
+from mathphys.beam_optics import beam_rigidity
+
+
+class Wire:
+    """Simulates the magnetic field of a infinite wire.
+    """
+    def __init__(self, position, current):
+        """ Initialize a Wire object.
+        Parameters
+        ----------
+        position : numpy.array of size 2
+            Cartesian coordinates [x,y] of the wire position in [m].
+
+        current : float, int
+            Current of the wire, in units of [A].
+        """
+        self.position = position  # [m]
+        self.current = current    # [A]
+
+    def calc_magnetic_field(self, r):
+        """Computes the magnetic field (in units of [T]) generated by the wire
+        in a set of points in space.
+
+        Parameters
+        ----------
+        r : numpy.array
+            Array of shape (2 x N) with the coordinates [[x1, x2, ... , xN],
+            [y1, y2, ... , yN]] where the field will be returned.
+
+        Returns
+        -------
+        numpy.array
+            Array with the same shape of r, containing the magnetic field
+            components at each point in r.
+
+        Raises
+        ------
+        ValueError
+            If input r hasn't shape of type (2 X N).
+        """
+        if r.shape[0] != 2:
+            raise ValueError(
+                "The input position vector r must have shape of type (2, N)")
+        r = r.reshape(2, -1)
+        mu0 = vacuum_permeability
+        r_w = self.position[:, None]  # Wire positions
+        rc = r - r_w  # Cursive r
+        theta = _np.arctan2(rc[1, :], rc[0, :])
+        theta_vec = _np.array([-_np.sin(theta), _np.cos(theta)])
+        rc_norm = _np.linalg.norm(rc, axis=0)[None, :]
+        mag_field = \
+            mu0 * self.current/(2*_np.pi*rc_norm)*theta_vec + r_w
+        return mag_field
+
+    @property
+    def position(self):
+        """."""
+        return self._pos
+
+    @position.setter
+    def position(self, position):
+        """."""
+        position = self._check_r_size(position)
+        self._pos = position
+
+    @property
+    def current(self):
+        """."""
+        return self._curr
+
+    @current.setter
+    def current(self, current):
+        """."""
+        self._curr = current
+
+    @staticmethod
+    def _check_r_size(r):
+        """."""
+        if r.size == 2:
```

Comment:
this function should also accept a list or a tuple as input. You can do this by adding a line
```
r = _np.asarray(r)
```
before the `if` statement.

Reply:
Done, thanks!